cartoframes/client/sql_client.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import absolute_import
+
+from .. import context
+
+
+class SQLClient(object):
+    """SQLClient class is a client to run SQL queries in a CARTO account.
+
+    Args:
+        credentials (:py:class:`Credentials <cartoframes.auth.Credentials>`):
+          A :py:class:`Credentials <cartoframes.auth.Credentials>`
+          instance can be used in place of a `username`/`api_key` combination.
+        session (requests.Session, optional): requests session. See `requests
+          documentation
+          <http://docs.python-requests.org/en/master/user/advanced/>`__
+          for more information.
+
+    Example:
+
+        .. code::
+
+            from cartoframes.auth import Credentials
+            from cartoframes.client import SQLClient
+
+            creds = Credentials(username='<USER NAME>', api_key='<API KEY>')
+            sql = SQLClient(creds)
+
+            sql.query('SELECT * FROM table_name')
+            sql.execute('DROP TABLE table_name')
+
+            sql.distinct('table_name', 'column_name')
+            sql.count('table_name')
+            ...
+    """
+
+    def __init__(self, credentials, session=None):
+        self._is_org_user = None
+        self._creds = credentials
+        self._context = context.create_context(credentials, session)
+
+    def query(self, query, verbose=False):
+        """Run a SQL query. It returns a `list` with content of the response.
+        If the `verbose` param is True it returns the full SQL response in a `dict`.
+        For more information check the `SQL API
+        documentation
+        <https://carto.com/developers/sql-api/reference/#tag/Single-SQL-Statement>`."""
```

Comment:
Maybe we could add a raises section and indicate that for `long running SELECT queries affected by database timeouts` it's recommended to use the `Dataset` class.